syntax = "proto3";

import "graph.proto";
import "map_renderer.proto";

package proto_tc;

message Coordinates {
    double        lat = 1;
    double        lon = 2;
}

message Stop {
    bytes         name = 1;
    Coordinates   coords = 2;
    bool          free = 3;
}

message Bus {
    bool               is_ring = 1;
    bytes              number = 2;
    repeated bytes     stop_names = 3;
}

message FromToDistance {
    bytes      from = 1;
    bytes      to = 2;
    uint64     distance = 3;
}

message TransportCatalogue {
    repeated Stop                 stops = 1;
    repeated FromToDistance       stops_distance = 2;
    repeated Bus                  buses = 3;
    proto_renderer.Map            map = 4;
    proto_tr.Settings             parameters = 5;
}

